compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp

```diff
@@ -62,8 +62,21 @@ using CodeGenPipeline = IREE::Codegen::DispatchLoweringPassPipeline;
 //===----------------------------------------------------------------------===//
 
 bool isMatmulOrBatchMatmul(linalg::LinalgOp linalgOp) {
-  return linalg::isaContractionOpInterface(linalgOp) &&
-         llvm::is_contained({2u, 3u}, linalgOp.getNumParallelLoops());
+  // (Batch) matmul should be a reduction op with 2/3 parallel dimensions.
+  if (!linalg::isaContractionOpInterface(linalgOp) ||
+      !llvm::is_contained({2u, 3u}, linalgOp.getNumParallelLoops()))
+    return false;
+
+  // Also exclude the case of matvec, which has only one non-unit parallel dim.
+  // They should go down different pipelines.
+  int nonUnitParallelDimCount = 0;
+  SmallVector<int64_t, 4> bounds = linalgOp.getStaticLoopRanges();
+  SmallVector<utils::IteratorType, 4> kinds = linalgOp.getIteratorTypesArray();
+  for (auto [kind, bound] : llvm::zip(kinds, bounds)) {
+    if (kind == utils::IteratorType::parallel)
+      nonUnitParallelDimCount += bound != 1;
+  }
+  return nonUnitParallelDimCount > 1;
```

Comment:
Actually I do not see this tested below (I only see a case with a single parallel dimension). It might be worth at least adding a configuration test.

Reply:
We have tests for both M=1 and (M != 1 && N != 1) cases. That should be enough?

Reply:
oh I missed the test in config_nvidia_matmul! That is plenty, just figured having the configuration somewhere would help to avoid surprises later.